src/Symfony/Component/HttpClient/Tests/AmpHttpClientTest.php

```diff
@@ -24,6 +30,27 @@ protected function getHttpClient(string $testCase): HttpClientInterface
         return new AmpHttpClient(['verify_peer' => false, 'verify_host' => false, 'timeout' => 5]);
     }
 
+    public function testUnixSocket()
+    {
+        if (\PHP_VERSION_ID < 80400) {
+            $this->markTestSkipped('Unix sockets only support PHP >= 8.4.');
+        }
+
+        if (strtoupper(substr(\PHP_OS, 0, 3)) === 'WIN') {
+            $this->markTestSkipped('Unix sockets are not supported on Windows.');
+        }
+
+        $client = new AmpHttpClient(['bindto' => '/var/run/docker.sock']);
+
+        $client = $client->withOptions(['base_uri' => 'http://docker']);
+
+        $response = $client->request('GET', '/info');
+
+        $this->assertSame(200, $response->getStatusCode());
```

Comment:
this test case is going to work only if a docker daemon is running, isn't it?
I guess we need to skip the test case if the `docker.sock` file does not exist (and thus remove the windows check, which would be redundant)